### ISSUE TITLE
Conformance to PDF/UA and PDF/A should not be mutually exclusive

### ISSUE THREAD

**Issue by moritzfl** [veraPDF/veraPDF-library#1456]
The PDF association states that PDF/A and PDF/UA are not mutially exclusive.
https://pdfa.org/pdf-standards-are-not-mutually-exclusive/

And while it may seem weird to include a schema for PDF/UA conformance within a PDF/A document, external schemata are indeed allowed by PDF/A and thus this can be used to achieve PDF/A and PDF/UA conformance.

This is relevant in practice as the European Accessibility Act dictates that every country in the EU must soon (next year) have solutions in place so that PDF/UA documents are provided for all customers that ask for accessible documents.

For archiving, it would be desirable to keep files that conform to PDF/UA while also ensuring PDF/A conformity.

An example for such a file that also passes Adobe Acrobats preflight checks is the following:
https://pdfa.org/wp-content/uploads/2018/03/Flyer-PDF-Association-PDFUA1-PDFA2u.pdf

It would be great if VeraPDF could cover this combination in the included checks accordingly. At the moment, such a file fails the VeraPDF check for PDF/A conformance.
![image2](https://github.com/veraPDF/veraPDF-library/assets/18644803/70190ef9-c2d4-4a2e-8637-60a3798d846a)
![image](https://github.com/veraPDF/veraPDF-library/assets/18644803/0c60397d-02c1-4bed-9d6b-fb2fdb1343a3)
<img width="1022" alt="image3" src="https://github.com/veraPDF/veraPDF-library/assets/18644803/81d0726e-9a87-49ed-ae4a-a26d6b600a0b">

**Comment by bdoubrov** [comment]
@moritzfl This seems to be a duplicate of #1414. Incidentally, just now we are testing the prototype implementation of this feature and here is the report for your PDF file, which is generated by this prototype. Is this what you would expect to have?
  
[htmlReport.zip](https://github.com/user-attachments/files/15740223/htmlReport.zip)

**Comment by moritzfl** [comment]
This is somewhat similar to the issue that I opened here but not exactly the same.

#1414 covers the case that conformance to two specifications is checked if a file claims to be a PDF/A and PDF/UA file at the same time.

This issue goes more towards the PDF/A and PDF/UA specification themselves.
In PDF/UA, XMP metadata does not have any real restriction. So the file I provided would pass a PDF/UA check (as also shown in the report that you posted).

However, the file does not pass a PDF/A conformance check as it includes metadata for PDF/UA that is not expected for PDF/A files.
And while this generally seems to make sense, the file does include the schema description for PDF/UA metadata in the XMP and thus should conform to PDF/A.

In other words - it should be possible to have a file that conforms to both PDF/A and PDF/UA at the same time and passes the checks in VeraPDF accordingly.

**Comment by bdoubrov** [comment]
There is no conflict between PDF/A and PDF/UA. However, in order to comply to both, the document must have all custom XMP properties defined by an appropriate extension schema as required by PDF/A. In your PDF example there are 4 properties (see the report) which are not a part of predefined schemas support by PDF/A-2:

`xmpTPg:HasVisibleTransparency`
`xmpTPg:HasVisibleOverprint`
`xmpTPg:SwatchGroups`
`illustrator:Type`

**Comment by moritzfl** [comment]
Ok - my bad.

Fixed the issues in the file and it now passes the PDF/A and PDF/UA conformance checks.

Its unfortunate that the files provided through the PDF Association to showcase mutual conformity to PDF/A and PDF/UA have such issues ...

Here is the fixed file for reference:
[Flyer-PDF-Association-PDFUA1-PDFA2u_fixed.pdf](https://github.com/user-attachments/files/15741460/Flyer-PDF-Association-PDFUA1-PDFA2u_fixed.pdf)

**Comment by DuffJohnson** [comment]
Thanks for highlighting (and fixing!) the problem with our file! 

This isn't the latest version of the "PDF Association Flyer" - it's an older one. The current version is [here](https://pdfa.org/resource/pdf-association-flyer/). I hope that it doesn't also have this problem... but it might.

Where did you find this old file linked on pdfa.org? We'd like to ensure that no-one downloads it again!

**Comment by moritzfl** [comment]
I believe, I downloaded the file from a link in this article (the file behind the "this document" link): https://pdfa.org/pdf-standards-are-not-mutually-exclusive/

I'll check the current version out on Monday and get back to you with the results.

**Comment by DuffJohnson** [comment]
Thank you!
At the risk of pissing off the Time Lords I've now corrected the link on that page to point to the latest version of the flyer. 
I look forward to your assesssment!

**Comment by moritzfl** [comment]
The current files still fail validation through VeraPDF.

I included a decoded version of the file for an easier analysis (pdf-association-flyer-decoded.pdf, I still like using Ctrl+F in a Text Editor sometimes). All object references mentioned in the following text refer to the decoded file. However, only the object numbers are different, the structure is pretty much the same as in the original file.

Object 1031 references the metadata object 1273 which contains the issues mentioned in the report (Report is also included in the zip-file.

Object 1031 is referenced as part of the Properties-Dictionary (as MC0 entry) from object 994 which is the Resources dictionary of the first page in the document.

Navigating through the structure from the first page of the PDF-Association-flyer.pdf document in a similar manner, you will find the same issue there (in PDFBox Debugger, the path to the metadata object with the issue would be Root/Pages/Kids/[0]/Resources/Properties/MC0/Metadata).

[PDF-Association-flyer-current-version.zip](https://github.com/user-attachments/files/15756722/PDF-Association-flyer-current-version.zip)
